### PR TITLE
breaking: require eslint v9 and switch to flat config

### PR DESCRIPTION
**Pull request by jrmajor**

### CODE REVIEW

**Comment by benmccann** [comment]
@jrmajor I see that this PR is currently marked as a draft. GitHub doesn't notify when a PR is goes from draft to ready for review, so please ping me when you're ready for me to take a look and I will review it

**Comment by benmccann** [comment]
Something else I was interested in trying was checking for promise-related errors. I just noticed in another project that TypeScript doesn't catch these types of errors. It should probably be another PR and isn't related to this change at all, but I thought I'd mention it just in case you were interested in working on it at all:

https://github.com/immich-app/immich/blob/7f504ec5fc6a300ec5265433a392f74db18c8e66/web/.eslintrc.cjs#L54

File: package.json [hunk 1]
```diff
@@ -19,24 +20,25 @@
...
-    "@typescript-eslint/eslint-plugin": ">= 5",
-    "@typescript-eslint/parser": ">= 5",
-    "eslint": ">= 8",
-    "eslint-config-prettier": ">= 8",
-    "eslint-plugin-svelte": ">= 2",
-    "eslint-plugin-unicorn": ">= 47",
-    "typescript": ">= 4"
+    "eslint": ">= 9",
+    "eslint-config-prettier": ">= 9",
+    "eslint-plugin-svelte": ">= 2.36",
```
Comment (benmccann) [inline package.json:37]: is there a reason we don't leave this as `>= 2`?

Comment (jrmajor) [reply package.json:37]: https://github.com/sveltejs/eslint-plugin-svelte/releases/tag/v2.36.0 added support for ESLint 9 / flat config.

File: index.js [hunk 2]
```diff
@@ -1,59 +1,63 @@
...
+	...ts.configs.recommended,
+	...svelte.configs['flat/recommended'],
+	prettier,
+	...svelte.configs['flat/prettier'],
+	{
+		plugins: {
+			unicorn
+		},
+		languageOptions: {
+			ecmaVersion: 13, // es2022
```
Comment (jrmajor) [inline index.js:20]: `ecmaVersion` now defaults to `latest`, so if that works for us we can just drop this line.

Comment (benmccann) [reply index.js:20]: Yes, I think using `latest` would be fine. I think we use the latest es features and then distribute svelte and sveltekit with the latest features and then the user's application will downgrade them to the targetted version typically with this vite setting, which defaults to es2020: https://vitejs.dev/config/build-options.html#build-target

**Review by benmccann — APPROVED** [review]

**Pull request merged**